dan_layer/consensus/src/block_validations.rs

```diff
@@ -42,6 +42,30 @@ pub fn check_proposed_by_leader<TAddr: DerivableFromPublicKey, TLeaderStrategy:
     Ok(())
 }
 
+pub fn check_signature(candidate_block: &Block) -> Result<(), ProposalValidationError> {
+    if candidate_block.is_dummy() {
```

Comment:
I think we should change the sync to never send and receive dummy blocks, We could calculate dummy blocks instead.